nomad/resource_node_pool.go

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package nomad
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-nomad/nomad/helper"
+	"github.com/hashicorp/terraform-provider-nomad/nomad/helper/pointer"
+)
+
+const (
+	nodePoolMemoryOversubscriptionEnabled  = "enabled"
+	nodePoolMemoryOversubscriptionDisabled = "disabled"
+)
+
+func resourceNodePool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNodePoolWrite,
+		Update: resourceNodePoolWrite,
+		Delete: resourceNodePoolDelete,
+		Read:   resourceNodePoolRead,
+		Exists: resourceNodePoolExists,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
```

Comment:
This is deprecated in favour of `ImportStatePassthroughContext`, do we want to use the non-deprecated function for this initial commit?

Reply:
Oh nice, I didn't notice the deprecation note, thanks!